app/javascript/packs/admin/editUser.jsx

```diff
@@ -1,8 +1,42 @@
 import { initializeDropdown } from '@utilities/dropdownUtils';
 
-let preact;
-let AdminModal;
-const modalContents = new Map();
+function adjustCreditRange(event) {
+  const {
+    target: { value, name, form },
+  } = event;
+
+  if (name === 'user[credit_action]') {
+    const creditAmount = form['user[credit_amount]'];
+
+    if (value === 'Add') {
+      if (creditAmount.getAttribute('data-old-max')) {
+        creditAmount.setAttribute(
+          'max',
+          creditAmount.getAttribute('data-old-max'),
+        );
+      }
+    } else {
+      creditAmount.setAttribute(
+        'data-old-max',
+        creditAmount.getAttribute('max'),
+      );
+      creditAmount.setAttribute('max', creditAmount.dataset.unspentCredits);
+    }
+  }
+}
+
+function enableEvents(key, enabled = true) {
```

Comment:
I was contemplating a disableEvents function as well, but for now, that seemed like overkill.